libcore/src/pedestrian/Pedestrian.cpp

```diff
@@ -446,35 +461,40 @@ double Pedestrian::GetV0Norm() const
     double nav_elevation = sub->GetElevation(target);
     double delta         = nav_elevation - ped_elevation;
     double walking_speed = 0;
+    auto subType         = sub->GetType();
+    double smoothFactor  = 15;
+    if(subType == "escalator" || subType == "escalator_up" || subType == "escalator_down" ||
+       subType == "idle_escalator_up" || subType == "idle_escalator_down") {
+        smoothFactor = _smoothFactorEscalator;
+    } else if(subType == "stair")
+        smoothFactor = _smoothFactorStair;
+
```

Comment:
When I looked at this code I was asking myself immediately: "Why ist smoothFactor set to 15?" Answer: well if `subtype` is not any of `escalator_...` or `stair`. Imagine you wrote something like:

```cpp
double Pedestrian::selectSmoothFactor(const std::string& type) const {
if(subType == "escalator" || subType == "escalator_up" || subType == "escalator_down" ||
        subType == "idle_escalator_up" || subType == "idle_escalator_down") {
         return _smoothFactorEscalator;
     }
     if(subType == "stair") {
         return = _smoothFactorStair;
     }
     // for unknown types we use the default
     return 15;
}
// and then later in your code
const double smoothFactor = selectSmoothFactor(subType);
```
This would even be nicer to read if the type of `subType` would be an `enum V0_Type`  and not a `std::string` In that case this could look like this:

```cpp
double Pedestrian::selectSmoothFactor(V0_Type type) const {
    switch(type) {
    case V0_Type::EscalatorUp:
    case V0_Type::EscalatorDown:
    case V0_Type::EscalatorIdleUp:
    case V0_Type::EscalatorIdleDown:
        return _smoothFactorEscalator;
    case V0_Type::StairUp:
    case V0_Type::StairDown:
        return _smoothFactorStair;
    case V0_Type::Unknown:
        return 15.0;
    }
}
```

By using a switch the compiler will warn you if you extend the enum with another case, e.g. "Spiral Stairs", if you do not handle this. (This is a warning that we have enabled afaik) But that might be way to much work right now.